rl4co/models/nn/attention.py

```diff
@@ -145,6 +146,7 @@ class PointerAttention(nn.Module):
         linear_bias: whether to use bias in linear projection
         check_nan: whether to check for NaNs in logits
         sdpa_fn: scaled dot product attention function (SDPA) implementation
+        moe_kwargs: Keyword arguments for MoE
     """
 
     def __init__(
```

Comment:
Mmmh... In principle, it might be better to have all the contributions of a model in its own repo. But given that such contribution could improve other models (say: if you wanted, you may try applying MoE to scheduling) and **if it does not change the base performance** (which seems to be the case here?), then I think it can be good. What do you think?

Asking also master @LTluttmann

Reply:
Yes, maybe I can create a new `class PointerAttention_MoE(PointerAttention)`. Let me know your final decision.

Reply:
I would also go for a separate `PointerAttnMoE(PointerAttention)` class for MoE, which would only need to overwrite the `project_out`. I guess we have the best of both worlds then - most of the `PointerAttention` code is reused while we separate the logic in a clear manner.

Reply:
okay, I will finish this part in the next commit!

Reply:
Hi all, I have finished this part. As for the MHA in `attnnet.py`, I think the current code is neat, so I keep it as is. In my opinion, MoE is a general idea, which may benefit various CO tasks. So implementing it in the underlying code may be a better choice..? What do you think? Kindly let me know if any problem exists.

Reply:
On that I agree, it doesn't impact performance in MHA in `attnnet.py` correct? If so, I'm good keeping it, so people may explore it!

Reply:
Yes, the default setting is `moe_kwargs=None`. In this case, it's equivalent to the original MHA, and thus the performance will not be affected.